### PR TITLE
[WNMGDS-1713] Date field docs

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary

[Jira ticket](https://jira.cms.gov/browse/WNMGDS-1713)

Create documentation based around [Confluence discussion](https://confluence.cms.gov/display/HCDSG/Date+input+feature#tab-Guidelines).

### To test

I tried building a demo site but I'm getting a 403 on the URLs.

To test locally, build both Core and Healthcare docs site (`yarn start` and `yarn start:healthcare`) and view the documentation for all three date field pages. The documentation should match Confluence's guidelines.

### CODE REVIEW

File: packages/ds-healthcare-gov/docs/src/pages/components/MultiInputDateField/_MultiInputDateField.docs.scss [hunk 1]
```diff
@@ -1,39 +1,18 @@
...
-*/
-
 /*
 ---
 
 ### When to use
 
-- Use this format for most dates, particularly memorized dates.
+- This date input option isn't recommended.
+- Use only when functionality/back-end systems can't support a single date field entry or if you must collect only month and year.
```
Comment (zarahzachz) [inline packages/ds-healthcare-gov/docs/src/pages/components/MultiInputDateField/_MultiInputDateField.docs.scss:7]: This was how it was written in Confluence, however we don't currently support the collection of month and year. Should I rephrase to:
> Use only when functionality/back-end systems can't support a single date field entry.

Comment (pwolfert) [reply packages/ds-healthcare-gov/docs/src/pages/components/MultiInputDateField/_MultiInputDateField.docs.scss:7]: Yeah, that's fine. I think that sentence could use some work, but this is what was approved.

File: packages/ds-healthcare-gov/docs/src/pages/components/MultiInputDateField/_MultiInputDateField.docs.scss [hunk 2]
```diff
@@ -46,8 +25,10 @@ Style guide: components.date-field.react
 
 ### Related patterns
 
+- [Date field (single-input)]({{root}}/components/single-input-date-field/)
+- [Date field (with calendar)]({{root}}/components/date-field-with-calendar-picker/)
```
Comment (pwolfert) [inline packages/ds-healthcare-gov/docs/src/pages/components/MultiInputDateField/_MultiInputDateField.docs.scss:29]: This link is broken on this page and another

File: packages/design-system-docs/src/pages/components/DatePicker/_CalendarPickerDateField.docs.scss [hunk 3]
```diff
@@ -0,0 +1,50 @@
+/*
+Date field (with calendar picker)
+
+@uswds https://designsystem.digital.gov/components/date-picker/
+
+Date field (with calendar picker), also known as _date picker_, provides a single date entry with a text field or selection through a visual calendar element tied to the input.
+
```
Comment (pwolfert) [inline packages/design-system-docs/src/pages/components/DatePicker/_CalendarPickerDateField.docs.scss:7]: I think in this introductory section at the top we should explain how the differentiator is the `to*` and `from*` props. In my latest branch, I've changed it to require both a `to*` and `from*` prop in order to turn it into a picker.

File: packages/design-system-docs/src/pages/components/DatePicker/_CalendarPickerDateField.docs.scss [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+/*
+---
+
+### When to use
+
+- Use the date picker for dates that have happened within the last year, in the future, or is relevant to selecting the day of the week.
+
+### When to consider alternatives
+
+- If users are entering a memorable date. Instead, use either single- or multi-input date fields.
```
Comment (pwolfert) [inline packages/design-system-docs/src/pages/components/DatePicker/_CalendarPickerDateField.docs.scss:30]: might be helpful to turn `single-` and `multi-input` into links to those component pages

**Review by pwolfert — COMMENTED** [review]
Guidance sections look good

**Review by forrestbaer — APPROVED** [review]
💯

**Review by scheul93 — APPROVED** [review]
Thanks for writing these! Good job differentiating between the different types

**Review by pwolfert — COMMENTED** [review]
🏁 🎉

**Pull request merged**